components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/LocalAndOutboundAuthenticationConfig.java

```diff
@@ -101,6 +103,14 @@ public static LocalAndOutboundAuthenticationConfig build(
                 if (member.getText() != null && "true".equals(member.getText())) {
                     localAndOutboundAuthenticationConfig.setAlwaysSendBackAuthenticatedListOfIdPs(true);
                 }
+            } else if ("useUserstoreDomainInUsername".equals(member.getLocalName())) {
+                if (member.getText() != null && "true".equals(member.getText())) {
```

Comment:
now the logic is changed to check for false. This is done to specifically set false from the xml. Otherwise it is considered as true. So cannot use parseBoolean as it will give false for anything other than "true"